azure-iot-device/azure/iot/device/common/pipeline/pipeline_stages_mqtt.py

```diff
@@ -59,13 +59,28 @@ def _execute_op(self, op):
             self.sas_token = op.sas_token
             self.client_cert = op.client_cert
 
-            self.transport = MQTTTransport(
-                client_id=self.client_id,
-                hostname=self.hostname,
-                username=self.username,
-                ca_cert=self.ca_cert,
-                x509_cert=self.client_cert,
-            )
+            if self.pipeline_root.pipeline_configuration:
+                self.transport = MQTTTransport(
+                    client_id=self.client_id,
+                    hostname=self.hostname,
+                    username=self.username,
+                    ca_cert=self.ca_cert,
+                    x509_cert=self.client_cert,
+                    websockets=self.pipeline_root.pipeline_configuration.websockets,
+                )
+            else:
+                # TODO:
+                # While MQTTWS in provisioning client is not implemented, this is defaulted
+                # to being false for websockets.
+                # When this is implemented there should be no IF here.
```

Comment:
This could be solved by always having the configuration in provisioning.

Reply:
yes, so when provisioning is implemented, it would always have the configuration and this if would not be necessary.

---
In reply to: [335713568](https://github.com/Azure/azure-iot-sdk-python/pull/331#discussion_r335713568) [](ancestors = 335713568)